plutus-tx/src/PlutusTx/Blueprint/Argument.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE RecordWildCards    #-}
+
+module PlutusTx.Blueprint.Argument where
+
+import Prelude
+
+import Data.Aeson (ToJSON (..), (.=))
+import Data.Aeson qualified as Aeson
+import Data.Maybe (catMaybes)
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Text (Text)
+import PlutusTx.Blueprint.Purpose (Purpose)
+import PlutusTx.Blueprint.Schema (Schema)
+
+data ArgumentBlueprint = MkArgumentBlueprint
+  { argumentTitle       :: Maybe Text
+  -- ^ A short and descriptive name for the redeemer or datum.
+  , argumentDescription :: Maybe Text
+  -- ^ An informative description of the redeemer or datum.
+  , argumentPurpose     :: Set Purpose
+  -- ^ A possibly empty set of purposes for the redeemer or datum.
+  , argumentSchema      :: Schema
+  -- ^ A Plutus Data Schema using the core vocabulary defined below,
+  -- or a oneOf applicator of Plutus Data Schemas
+  }
+  deriving stock (Show)
+
+instance ToJSON ArgumentBlueprint where
```

Comment:
surely we're going to want the other direction too? Another point in favour of generating it if we can.

Reply:
Actually I am not sure if we're going to ever implement parsing blueprints so I have intentionally omitted `FromJSON` instances following the [YAGNI principle](https://martinfowler.com/bliki/Yagni.html).

Reply:
If anyone was going to _use_ blueprints written by someone else to operate a contract they would want this. It seems bizarre to me for us to provide a library for working with these and to not provide a way to parse them.

Reply:
The only reason it costs us anything to do is because we're not deriving all these instances ;)

Reply:
This particular instance has custom encoding logic for the `purpose` field, which I don't know how to implement using an instance derivation without leaking a derivation-specific newtypes into the user space, so I decide to leave it manual.

Reply:
> It seems bizarre to me for us to provide a library for working with these and to not provide a way to parse them.

Yes, that seems a bit strange.  It would make testing a lot easier if we had a parser.  It might be a lot of work if we're not generating instances automatically though.